samples/Mvc.RenderViewToString/Program.cs

```diff
@@ -0,0 +1,54 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Diagnostics;
+using Microsoft.AspNet.FileProviders;
+using Microsoft.AspNet.Hosting;
+using Microsoft.AspNet.Hosting.Internal;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.PlatformAbstractions;
+
+namespace Mvc.RenderViewToString
+{
+    public class Program
+    {
+        public static void Main()
+        {
+            // Initialize the necessary services
+            var services = new ServiceCollection();
+            ConfigureDefaultServices(services, path: null);
+
+            // Add a custom service that is used in the view.
+            services.AddSingleton<EmailReportHelper>();
```

Comment:
If that's interesting, suggest using a more mainline scenario for `@inject` then e.g. a `ViewLocalizer`. But that might add to many unrelated complications to the sample. And MVC uses `@inject` internally so you're not really testing anything new except this helper itself.